docs/docs/endpoints.md

```diff
@@ -64,3 +64,34 @@ of the parsed `path`, `query`, and `body` parameters squashed together, a `conte
 or resolve to a raw response to be parsed by the `response` schema.
 
 The `params`, `context` and return types will be automatically mapped from the request and response schemas.
+
+## Error handling
+
+In order to indicate an unauthorized usage of or error encountered during the 
+handling of an endpoint, throw an exception within the `handler`.
```

Comment:
```suggestion
You can throw certain exceptions within the `handler` to return 4xx or 5xx HTTP error responses to the user, with well-formatted bodies.
```

Reply:
Kind of a side question, but do you see 3xx status codes as something we should support? The exception mechanism is probably not the right one, at least for some codes in the group.

Reply:
My instinct is no, API routes should not typically 3xx, but I may be wrong about that. I can't immediately think of a case where this would make sense, so I'm not sure whether raising or returning to 3xx would make more sense.